src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Test-Json command.
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Test, "Json", HelpUri = "")]
+    public class TestJsonCommand : PSCmdlet
+    {
+        /// <summary>
+        /// An Json to be validated.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public String Json { get; set; }
+
+        /// <summary>
+        /// A schema to validate the Json against.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        public String Schema { get; set; }
+
+        private JSchema _jschema = null;
+
+        /// <summary>
+        /// Prepare an Json schema.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            if (Schema != null)
+            {
+                try
+                {
+                    _jschema = JSchema.Parse(Schema);
+                }
+                catch (Exception exc)
+                {
+                    Exception exception = new Exception(TestJsonCmdletStrings.InvalidJsonSchema, exc);
+                    ThrowTerminatingError(new ErrorRecord(exception, "InvalidJsonSchema", ErrorCategory.InvalidData, null));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validate an Json.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            JObject parsedJson = null;
+            bool result = true;
+
+            try
+            {
+                parsedJson = JObject.Parse(Json);
```

Comment:
I am not familiar with this framework.  Do we need to be concerned about code injection here when passing the input string directly to the Parse() API?

Reply:
Good question.
@JamesNK Could you please comment? Should we be concerned about code injection in `JObject.Parse(Json)` ?

Reply:
We can look at the [Parse source code](https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Linq/JObject.cs#L423). It uses `StringReader` at low level and then tokenize the string stream. 
I don't see threats but I haven't an experience in the security area.

Reply:
I agree that it is an unlikely threat with this Json reader/parser, but I wanted to raise awareness for possible code injection vulnerabilities when we pass external input to a 3rd party API.